core/tests/test_suite.py

```diff
@@ -0,0 +1,114 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Oppia-ml test suite.
```

Comment:
The PR itself is small. It seems big because of testing related files (like `test_utils.py`, `test_suite.py`, `backend_test.py` and they are mostly copy of Oppia's files). So the main logic of remote_services is pretty small.

I understand that it might have been better if I set up testing framework in a separate PR but the kind of stubs and test functions that will be required for testing became clear to me after implementing the actual functions.

So I first implemented code and then used stubs and added test utility functions. But the main purpose of PR is to implement remote services functions and that part is not too big. So I don't think there should be much trouble reviewing this. I apologize if it is harder then it could have been.